apps/prairielearn/src/lib/score-upload.js

```diff
@@ -9,372 +9,387 @@ const manualGrading = require('./manualGrading');
-    const course_instance_id = result.rows[0].course_instance_id;
-    const course_id = result.rows[0].course_id;
-
-    const serverJob = await createServerJob({
-      courseId: course_id,
-      courseInstanceId: course_instance_id,
-      assessmentId: assessment_id,
-      userId: user_id,
-      authnUserId: authn_user_id,
-      type: 'upload_instance_question_scores',
-      description: 'Upload question scores for ' + assessment_label,
+    const csvConverter = csvtojson({
+      colParser: {
+        instance: 'number',
+        score_perc: 'number',
+        points: 'number',
+        manual_score_perc: 'number',
+        manual_points: 'number',
+        auto_score_perc: 'number',
+        auto_points: 'number',
+        submission_id: 'number',
+      },
+      maxRowLength: 10000,
     });
 
-    serverJob.executeInBackground(async (job) => {
-      job.info('Uploading question scores for ' + assessment_label);
-
-      // accumulate output lines in the "output" variable and actually
-      // output put them in blocks, to avoid spamming the updates
-      let output = null;
-      let outputCount = 0;
-      let outputThreshold = 100;
-
-      let successCount = 0;
-      let errorCount = 0;
-
-      job.info(`Parsing uploaded CSV file "${csvFile.originalname}" (${csvFile.size} bytes)`);
-      const csvStream = streamifier.createReadStream(csvFile.buffer, {
-        encoding: 'utf8',
-      });
-      const csvConverter = csvtojson({
-        colParser: {
-          instance: 'number',
-          score_perc: 'number',
-          points: 'number',
-          manual_score_perc: 'number',
-          manual_points: 'number',
-          auto_score_perc: 'number',
-          auto_points: 'number',
-          submission_id: 'number',
-        },
-        maxRowLength: 10000,
-      });
-
-      try {
-        await csvConverter.fromStream(csvStream).subscribe(async (json, number) => {
-          // Replace all keys with their lower-case values
-          json = _.mapKeys(json, (_v, k) => k.toLowerCase());
-          const msg = `Processing CSV record ${number}: ${JSON.stringify(json)}`;
-          if (output == null) {
-            output = msg;
-          } else {
-            output += '\n' + msg;
-          }
-          try {
-            await module.exports._updateInstanceQuestionFromJson(
-              json,
-              assessment_id,
-              authn_user_id,
-            );
+    try {
+      await csvConverter.fromStream(csvStream).subscribe(async (json, number) => {
+        // Replace all keys with their lower-case values
+        json = _.mapKeys(json, (_v, k) => k.toLowerCase());
+        try {
+          if (await _updateInstanceQuestionFromJson(json, assessment_id, authn_user_id)) {
             successCount++;
-          } catch (err) {
-            errorCount++;
-            const msg = String(err);
+            const msg = `Processed CSV record ${number}: ${JSON.stringify(json)}`;
```

Comment:
One comment about this number: the number here is zero-based, which means that record 0 would be the first data line (second line in the file if you count the header). I wonder if this could be confusing for non-tech instructors, and if it may be a better idea to add 2 to the number to represent the line number instead (in essence "Processed line 2" instead of "Processed record 0"). Alternatively both could be shown, but I don't think that's a good idea. Thoughts?

Reply:
I think talking in terms of 1-indexed lines makes a lot of sense here, as that will match 1:1 with line/row numbers in an editor or Excel. I'd be happy with that change!

Reply:
Done in 6d11c24098d7299aa2160ef242f1f47a68c92e88. I also did the same for assessment instance upload.